c++/src/capnp/rpc.c++

```diff
@@ -2916,11 +2863,15 @@ private:
 
       pipelineToRelease = kj::mv(answer->pipeline);
 
-      // If the call isn't actually done yet, cancel it.  Otherwise, we can go ahead and erase the
-      // question from the table.
       KJ_IF_MAYBE(context, answer->callContext) {
-        context->requestCancel();
+        // Destroying answer->task will probably destroy the call context, but we can't prove that
+        // since it's refcounted. Instead, inform the call context that it is now its job to
+        // clean up the answer table. Then, cancel the task.
```

Comment:
Well... _technically_ it's correct because `promiseToRelease` is not dropped until the end of the function. :) Anyway I doubt anyone will be confused.